### PR TITLE
docs: Clarify `prometheus_remote_write` usage

### PR DESCRIPTION
**Pull request by jszwedko**
Users have been, understandably, confused thinking that this sink could
be used to send data to Prometheus itself.

**EDIT** Turns out this can be used to send directly to Prometheus now if you enable the feature via `--enable-feature=remote-write-receiver`.

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

File: docs/cue/reference/services/prometheus_remote_write.cue [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+services: prometheus_remote_write: {
+	name:     "Prometheus Remote Write Integration"
+	thing:    "a metrics database or service"
+	url:      urls.prometheus_remote_integrations
+	versions: null
+
+	description: """
+		Databases and services that are capable of receiving data via the Prometheus
+		[`remote_write`](\(urls.prometheus_remote_write_protocol)).
```
Comment (bruceg) [inline docs/cue/reference/services/prometheus_remote_write.cue:11]:
```suggestion
		[`remote_write` protocol](\(urls.prometheus_remote_write_protocol)).
```

**Review by bruceg — CHANGES_REQUESTED** [review]

**Review by spencergilbert — APPROVED** [review]
I think the healthcheck is perhaps not optimal still, or needs to be configurable due to the number of targets available for remote_write

**Comment by jszwedko** [comment]
> I think the healthcheck is perhaps not optimal still, or needs to be configurable due to the number of targets available for remote_write

Yeah, that's true. There doesn't seem to be a healthcheck defined for the protocol itself. We may just want to simply check for network connectivity. I'll opened https://github.com/timberio/vector/issues/8279 for that.

File: docs/cue/reference/services/prometheus_remote_write.cue [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	name:     "Prometheus Remote Write Integration"
+	thing:    "a metrics database or service"
+	url:      urls.prometheus_remote_integrations
+	versions: null
+
+	description: """
+		Databases and services that are capable of receiving data via the Prometheus
+		[`remote_write protocol`](\(urls.prometheus_remote_write_protocol)).
+
+		Prometheus itself can also receive metrics via the `remote_write` protocol if the feature is enabled via
```
Comment (bruceg) [inline docs/cue/reference/services/prometheus_remote_write.cue:13]: For clarification, does this feature need to be enabled at compile time (ie `configure --enable…`) or at run time?

Comment (spencergilbert) [reply docs/cue/reference/services/prometheus_remote_write.cue:13]: It's a runtime flag:

```
docker run --rm -it prom/prometheus --enable-feature=remote-write-receiver
...
level=info ts=2021-07-16T12:47:42.623Z caller=main.go:149 msg="Experimental remote-write-receiver enabled"
```

**Review by bruceg — APPROVED** [review]
LGTM except for possibly clarifying the enable option usage.

**Comment by jszwedko** [comment]
Doh, I had enabled the auto squash. I'll open a follow up PR to clarify the flag.

**Pull request merged**